src/core/OSTreeTUI.hpp

```diff
@@ -3,20 +3,139 @@
  |   A terminal user interface for OSTree.
  |___________________________________________________________*/
 
+#pragma once
+
+#include <memory>
 #include <string>
 #include <vector>
 
+#include "ftxui/component/component.hpp"  // for Renderer, ResizableSplitBottom, ResizableSplitLeft, ResizableSplitRight, ResizableSplitTop
+#include "ftxui/component/component_base.hpp"      // for ComponentBase
+#include "ftxui/component/screen_interactive.hpp"  // for ScreenInteractive
+#include "ftxui/dom/elements.hpp"  // for Element, operator|, text, center, border
+
+#include "commit.hpp"
+#include "footer.hpp"
+#include "manager.hpp"
+
 #include "../util/cpplibostree.hpp"
 
-namespace OSTreeTUI {
+class OSTreeTUI {
+public:
+    /**
+     * @brief Constructs, builds and assembles all components of the OSTreeTUI.
+     * 
+     * @param repo ostree repository (OSTreeRepo)
+     * @param startupBranches optional list of branches to pre-select at startup (providing nothing will display all branches)
```

Comment:
```suggestion
     * @param startupBranches Optional list of branches to pre-select at startup (providing nothing will display all branches).
```

Comments should always start with an upper case character and be be a full sentence. In this case a full sentence is not strictly necessary but in general it helps you to avoid comments where you do not answer "How?" and "Why?".